plugin-server/src/ingestion/ingestion-consumer.test.ts

```diff
@@ -0,0 +1,265 @@
+                join: () => ({
+                    finally: jest.fn(),
+                }),
+                stop: jest.fn(),
+                consumer: mockConsumer,
+            })
+        ),
+    }
+})
+
+jest.setTimeout(1000)
+
+let offsetIncrementer = 0
+
+const createKafkaMessages: (events: PipelineEvent[]) => Message[] = (events) => {
+    return events.map((event) => {
+        // TRICKY: This is the slightly different format that capture sends
+        const captureEvent = {
+            uuid: event.uuid,
+            distinct_id: event.distinct_id,
+            ip: event.ip,
+            now: event.now,
+            token: event.token,
+            data: JSON.stringify(event),
+        }
+        return {
+            value: Buffer.from(JSON.stringify(captureEvent)),
+            size: 1,
+            topic: 'test',
+            offset: offsetIncrementer++,
+            partition: 1,
+        }
+    })
+}
+
+describe('IngestionConsumer', () => {
+    let ingester: IngestionConsumer
+    let hub: Hub
+    let team: Team
+    let team2: Team
+
+    const createEvent = (event?: Partial<PipelineEvent>): PipelineEvent => ({
+        distinct_id: 'user-1',
+        uuid: new UUIDT().toString(),
+        token: team.api_token,
+        ip: '127.0.0.1',
+        site_url: 'us.posthog.com',
+        now: DateTime.now().toISO(),
+        event: '$pageview',
+        properties: {
+            $current_url: 'http://localhost:8000',
+        },
+        ...event,
+    })
+
+    beforeEach(async () => {
+        const now = new Date(2025, 1, 1).getTime()
+        jest.spyOn(Date, 'now').mockReturnValue(now)
+        offsetIncrementer = 0
+        await resetTestDatabase()
+        hub = await createHub()
+
+        hub.kafkaProducer = mockProducer
+        team = await getFirstTeam(hub)
+        const team2Id = await createTeam(hub.db.postgres, team.organization_id)
+        team2 = (await hub.db.fetchTeam(team2Id)) as Team
+    })
+
+    afterEach(async () => {
+        jest.restoreAllMocks()
+        jest.setTimeout(10000)
+        if (ingester) {
+            await ingester.stop()
+        }
+        await closeHub(hub)
+    })
+
+    afterAll(() => {
+        jest.useRealTimers()
+    })
+
+    describe('general', () => {
+        beforeEach(async () => {
+            ingester = new IngestionConsumer(hub)
+            await ingester.start()
+        })
+
+        it('should have the correct config', () => {
+            expect(ingester['name']).toMatchInlineSnapshot(`"ingestion-consumer-events_plugin_ingestion_test"`)
+            expect(ingester['groupId']).toMatchInlineSnapshot(`"events-ingestion-consumer"`)
+            expect(ingester['topic']).toMatchInlineSnapshot(`"events_plugin_ingestion_test"`)
+            expect(ingester['dlqTopic']).toMatchInlineSnapshot(`"events_plugin_ingestion_dlq_test"`)
+            expect(ingester['overflowTopic']).toMatchInlineSnapshot(`"events_plugin_ingestion_overflow_test"`)
+        })
```

Comment:
note: Not a huge fan of this kind of tests – they don't really check any behaviours. We still have to test all of the functionalities that use the config and the code could still reference the wrong config variable, which wouldn't be caught here.

Reply:
Fair enough. I personally am a fan of some lightweight testing like this as it gives a clear indication of what is broken and having these values be correct is very important. Gonna leave it for now but ill think about it as i continue